cargo-near-build/src/near/build_extended/mod.rs

```diff
@@ -1,54 +1,52 @@
 mod build_script;
-mod tmp_env;
+mod tmp_change_cwd;
 use crate::types::near::build::output::CompilationArtifact;
 use crate::types::near::build_extended::OptsExtended;
+use crate::{BuildImplicitEnvOpts, BuildOpts};
 use rustc_version::Version;
 
+use crate::extended::BuildScriptOpts;
+
+/// only single-threaded build-scripts are supported
+///
+/// this function cannot be run concurrently with itself, as it changes working dir
+/// of process, executing it.
+/// changing current working dir concurrently from different threads
+/// may entail incorrect results
```

Comment:
Can we resolve this limitation? E.g. specify it as an explicit command line argument to `cargo build`, or [set it as part of `exec`](https://doc.rust-lang.org/stable/std/process/struct.Command.html#method.current_dir). I believe that the current implementation may cause problems with tests and other concurrent runs that will be hard to debug.